mongodbatlas/search.go

```diff
@@ -0,0 +1,233 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const (
+	searchBasePath = "groups/%s/clusters/%s/fts"
+)
+
+// SearchService provides access to the search related functions in the Atlas API.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/atlas-search/
+type SearchService interface {
+	ListIndexes(context.Context, string, string, string, string, *ListOptions) ([]*SearchIndex, *Response, error)
+	GetIndex(context.Context, string, string, string) (*SearchIndex, *Response, error)
+	CreateIndex(context.Context, string, string, *SearchIndex) (*SearchIndex, *Response, error)
+	UpdateIndex(context.Context, string, string, string, *SearchIndex) (*SearchIndex, *Response, error)
+	DeleteIndex(context.Context, string, string, string) (*Response, error)
+	ListAnalyzers(context.Context, string, string, *ListOptions) ([]*SearchAnalyzer, *Response, error)
```

Comment:
Curious if leaving Update Analyzers out was intentional? (https://docs.atlas.mongodb.com/reference/api/fts-analyzers-update-all/)

Reply:
Somewhat intentional, there are plans to deprecate Custom Analyzers so I'm trying to limit us to only read them (if ever needed)